firmware/src/network_model.c

```diff
@@ -0,0 +1,829 @@
+  }
+
+  if (interface_id != NO_DEFAULT_ROUTE &&
+      (LookupIndex(interface_id) < 0 || interface_id != IPSEC_INTERFACE_ID)) {
+    // Only the ptp IPsec interface can be used as the default route
+    return RDMResponder_BuildNack(header, NR_DATA_OUT_OF_RANGE);
+  }
+  g_network_model.default_interface_route = interface_id;
+  g_network_model.default_route = ip;
+  return RDMResponder_BuildSetAck(header);
+}
+
+int NetworkModel_GetNameServer(const RDMHeader *header,
+                               const uint8_t *param_data) {
+  if (header->param_data_length != sizeof(uint8_t)) {
+    return RDMResponder_BuildNack(header, NR_FORMAT_ERROR);
+  }
+
+  uint8_t index = param_data[0];
+  if (index >= NUMBER_OF_NAMESERVERS) {
+    return RDMResponder_BuildNack(header, NR_DATA_OUT_OF_RANGE);
+  }
+
+  ReturnUnlessUnicast(header);
+  uint32_t ip = g_network_model.nameservers[index];
+
+  unsigned int offset = sizeof(RDMHeader);
+  g_rdm_buffer[offset++] = index;
+  g_rdm_buffer[offset++] = UInt32Byte0(ip);
+  g_rdm_buffer[offset++] = UInt32Byte1(ip);
+  g_rdm_buffer[offset++] = UInt32Byte2(ip);
+  g_rdm_buffer[offset++] = UInt32Byte3(ip);
+  RDMResponder_BuildHeader(header, ACK, GET_COMMAND_RESPONSE,
+                           ntohs(header->param_id),
+                           offset - sizeof(RDMHeader));
+  return RDMUtil_AppendChecksum(g_rdm_buffer);
+}
+
+int NetworkModel_SetNameServer(const RDMHeader *header,
+                               const uint8_t *param_data) {
+  if (header->param_data_length != sizeof(uint8_t) + sizeof(uint32_t)) {
+    return RDMResponder_BuildNack(header, NR_FORMAT_ERROR);
+  }
+
+  uint8_t index = param_data[0];
+  if (index >= NUMBER_OF_NAMESERVERS) {
+    return RDMResponder_BuildNack(header, NR_DATA_OUT_OF_RANGE);
+  }
+
+  uint32_t ip = JoinUInt32(param_data[1], param_data[2],
+                           param_data[3], param_data[4]);
+
+  g_network_model.nameservers[index] = ip;
+  return RDMResponder_BuildSetAck(header);
+}
+
+int NetworkModel_GetHostname(const RDMHeader *header,
+                             UNUSED const uint8_t *param_data) {
+  return RDMResponder_GenericReturnString(header, g_network_model.hostname,
+                                          DNS_HOST_NAME_SIZE);
+}
+
+int NetworkModel_SetHostName(const RDMHeader *header,
+                             const uint8_t *param_data) {
+  if (header->param_data_length == 0 ||
+      header->param_data_length > DNS_HOST_NAME_SIZE) {
+    return RDMResponder_BuildNack(header, NR_DATA_OUT_OF_RANGE);
+  }
+
+  RDMUtil_StringCopy(g_network_model.hostname, DNS_HOST_NAME_SIZE,
+                     (const char*) param_data,
+                     header->param_data_length);
+  return RDMResponder_BuildSetAck(header);
+}
+
+int NetworkModel_GetDomainName(const RDMHeader *header,
+                               UNUSED const uint8_t *param_data) {
+  return RDMResponder_GenericReturnString(header, g_network_model.domain_name,
```

Comment:
The name can't exceed 231 because that's all the space we have to store it. The HARDWARE_FAULT occurs when the name is set by some other mechanism (CLI or the like).